contrib/verify-commits/verify-commits.sh

```diff
@@ -123,9 +125,29 @@ while true; do
 	fi
 
 	PARENTS=$(git show -s --format=format:%P "$CURRENT_COMMIT")
-	for PARENT in $PARENTS; do
-		PREV_COMMIT="$CURRENT_COMMIT"
-		CURRENT_COMMIT="$PARENT"
-		break
-	done
+	PARENT1=${PARENTS%% *}
+	PARENT2=""
+	if [ "x$PARENT1" != "x$PARENTS" ]; then
+		PARENTX=${PARENTS#* }
+		PARENT2=${PARENTX%% *}
+		if [ "x$PARENT2" != "x$PARENTX" ]; then
+			echo "Commit $CURRENT_COMMIT is an octopus merge" > /dev/stderr
+			exit 1
+		fi
+	fi
+	if [ "x$PARENT2" != "x" ]; then
+		CURRENT_TREE="$(git show --format="%T" "$CURRENT_COMMIT")"
+		git checkout --force --quiet "$PARENT1"
```

Comment:
Wouldn't a checkout also checkout the other files in `./contrib/verify-commit`, such as `trusted-git-root` or `trusted-keys` and thus change them under its own feet?